keps/1145-additional-labels/README.md

```diff
@@ -0,0 +1,153 @@
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Allow (AdmissionCheck)[https://github.com/kubernetes-sigs/kueue/tree/main/keps/993-two-phase-admission] 
+to set some additional location/admission information
+on workload's pods, so that they land on the machines of the `AdmissionCheck`
+choosing.
+
+## Motivation
+
+`AdmissionChecks` is a "plugin" mechanism for Kueue. Some of the plugins while 
+admitting may either narrow down on which machines the workload should start (by adding
+some extra node selector/tolerations) or add some tracking information
+that may be needed by other system components outside of Kueue.
+
+### Goals
+
+* Establish a mechanism through which `AdmissionChecks` can add annotations/labels/node selector/tolerations
+  to workload pods.
+
+### Non-Goals
+
+* Allow `AdmissionChecks` to change pod definition in other way, for instance change command or image name.
+* Hard code any specific labels/annotations that are passed down.
+
+## Proposal
+
+* Add a struct in `Workload` status that allows to set annotations, labels, selectors and tolerations on PodSets.
+* Expend PodSetInfo in Integration Framework to pass this information.
+* Modify existing integrations to support setting more data on the pods.
+
+### User Stories (Optional)
+
+#### Story 1
+
+My external controller managing an `AdmissionCheck` preallocates specific
+nodes (tainted and labeled) for a Workload to run, before it is 
+admitted (and creates any worker pods). Workload pods need to be modified
+so that they end up on correct nodes.
+
+#### Story 2
+
+My `AdmissionCheck` does budget checking and would like to mark pods that are possibly crossing budget with a 
+specific label for easier tracking and alerting.
+
+### Risks and Mitigations
+
+Some integrations may not support setting all of the desired information due to their CRD limitations.
+
+## Design Details
+
+Modify WorkloadStatus API object:
+
+```
+// WorkloadStatus defines the observed state of Workload
+type WorkloadStatus struct {
+    [...]
+    AdmissionChecks []metav1.Condition `json:"admissionChecks,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+    PodSetUpdates []PodSetUpdate `json:"podSetUpdates,omitempty"
+}
+
+// PodSetUpdate contains a list of pod set modifications suggested by AdmissionChecks.
+// The modifications should be additive only - modifications of already existing keys
+// are not allowed and will result in failure during workload admission.
```

Comment:
Some questions, potentially points to clarify: 
1. failing on conflicting keys does not seem consistent with what we do for nodeSelectors, where we overwrite (based on how the `MergeKeepFirst` is used): https://github.com/kubernetes-sigs/kueue/blob/f3bb6f2b6e509727e33b4aac84d2503cc9edaa0a/pkg/controller/jobs/job/job_controller.go#L228. 
2. Should we fail admission if there are tolerations with the same key, but different operator, or effect? It seems there might be legitimate use cases for adding an annotation in PodSetUpdate that has the same key, but different effect.

Reply:
For now, we should fail the workload completely if there is a conflict in node selectors/labels/etc. At least for now. If we get some feedback that conflict resolution is needed we can add it later as a non breaking change.